### DISCUSSION TITLE
Budibase tables - Attachment API - get file by file name

### DISCUSSION THREAD

**Discussion by luigiPinnaAevo** [Budibase/budibase#12507] [Help] [answered]
Hi guys,

I need help getting a specific file in the budibase table by API. I have a budibase table with "Attachment", "FileName" and other properties. My document table has about 16k rows. Due to the limit of 1000 rows per query, I need to get only one file filtering, for example, by fileName. Is it possible?

Thanks

**Comment by ConorWebb96** [comment, marked as answer]
Hey @luigiPinnaAevo,

I haven't tested this with 16k rows but this should be possible with the aid of a static formula column.

1. Here is the structure of my table with the formula column
<img width="100%" alt="Screenshot 2023-12-06 at 10 20 30" src="https://github.com/Budibase/budibase/assets/126772285/6a2169ad-9af5-4199-80af-92eb8220d282">
2. Here is the setup of the formula column
<img width="100%" alt="Screenshot 2023-12-06 at 10 21 24" src="https://github.com/Budibase/budibase/assets/126772285/4022e3a0-d6a1-4883-81b8-768e86af7b9a">
<img width="100%" alt="Screenshot 2023-12-06 at 10 21 30" src="https://github.com/Budibase/budibase/assets/126772285/dfc2aad0-f536-4940-b790-320da16efe69">
3. Here is the setup of the public api from the [docs](https://docs.budibase.com/reference/rowsearch)
<img width="100%" alt="Screenshot 2023-12-06 at 10 22 59" src="https://github.com/Budibase/budibase/assets/126772285/fd1fc761-544a-46bc-a2af-fcc620258437">
4. Here is the results from the public api showing it working and returning the correct record back whenever searching for the correct column value.
<img width="100%" alt="Screenshot 2023-12-06 at 10 33 49" src="https://github.com/Budibase/budibase/assets/126772285/6abb424a-b525-4542-87e4-69332f5cbc1d">

I hope this points you in the right direction.

**Comment by luigiPinnaAevo** [comment]
Ok great, thank you very much for the support @ConorWebb96 . I have only a doubt about the point 3. Where I can find that: _3. Here is the setup of the public API from the docs_?

**Reply by ConorWebb96** [reply]
Apologies here is the docs for the public api. https://docs.budibase.com/reference/rowsearch

**Reply by luigiPinnaAevo** [reply]
Thanks @ConorWebb96 I solved the issue! 👍👍
